### ISSUE TITLE
Preprocessing for other datasets

### ISSUE THREAD

**Issue by Miki-3** [nyu-dl/dl4mt-c2c#14]
Much thanks for sharing your code.
Now  I want to use the model for other datasets, such as chinese-english. I know I need to perform the preprocess.sh script. I'm using yours WMT'15 preprocessed corpora at the moment, but I don't know how to process the datasets ( including training datas and dictionaries ) into the following files. In particular, I don't know how the three dictionaries are generated separately.

/home/miki/dl4mt-c2c-master/deen/models/de_en/ bi-bpe2char
/home/miki/dl4mt-c2c-master/deen/train/all_de-en.de.tok.bpe.shuf
/home/miki/dl4mt-c2c-master/deen/train/all_de-en.en.tok.shuf
/home/miki/dl4mt-c2c-master/deen/dev/newstest2013.de.tok.bpe
/home/miki/dl4mt-c2c-master/deen/dev/newstest2013.en.tok
/home/miki/dl4mt-c2c-master/deen/train/all_de-en.de.tok.bpe.word.pkl
/home/miki/dl4mt-c2c-master/deen/train/all_de-en.en.tok.300.pkl

Another, what is the form of pure character input? The file all_de-en.de.tok.shuf is the input, of char2char, why not enter pure character training datas?
Thanks very much!

**Comment by jaseleephd** [comment]
.bpe files are tokenised using the Byte Pair Encoding algorithm (Sennrich et al., 2015)
.tok files are tokenised input using `tokenizer.perl` (taken from Moses).
.shuf files are shuffled in random order (same ordering maintained between source & target)

To train a char2char model you need none of the above, although you do need to extract vocabulary using `build_dictionary_char.py`. Treatment of Chinese can be important, if you use pinyin you're probably fine with standard hyperparameters (number of convolutional filters, filter width)

If you're not using pinyin (thereby using original Chinese characters), you should probably use a larger number of filters. I recommend starting out with pinyin.

**Comment by DivyaRathod3D** [comment]
How to proceed further after successfully building the vocabulary using ```build_dictionary_char.py```? I just want to use the character-level NMT for Hindi-English. Can you also tell me what all files shall be of my concern?
